test/get.js

```diff
@@ -17,44 +16,44 @@ test('get', function() {
          TypeError,
          'Invalid value\n' +
          '\n' +
-         'get :: Accessible a => TypeRep -> String -> a -> Maybe b\n' +
-         '                       ^^^^^^^\n' +
+         'get :: Accessible a => Function -> String -> a -> Maybe c\n' +
+         '                       ^^^^^^^^\n' +
          '                          1\n' +
          '\n' +
          '1)  [1, 2, 3] :: Array Number, Array FiniteNumber, Array NonZeroFiniteNumber, Array Integer, Array ValidNumber\n' +
          '\n' +
-         'The value at position 1 is not a member of ‘TypeRep’.\n');
+         'The value at position 1 is not a member of ‘Function’.\n');
 
-  throws(function() { S.get(Number, [1, 2, 3]); },
+  throws(function() { S.get(S.K(true), [1, 2, 3]); },
          TypeError,
          'Invalid value\n' +
          '\n' +
-         'get :: Accessible a => TypeRep -> String -> a -> Maybe b\n' +
-         '                                  ^^^^^^\n' +
-         '                                    1\n' +
+         'get :: Accessible a => Function -> String -> a -> Maybe c\n' +
+         '                                   ^^^^^^\n' +
+         '                                     1\n' +
          '\n' +
          '1)  [1, 2, 3] :: Array Number, Array FiniteNumber, Array NonZeroFiniteNumber, Array Integer, Array ValidNumber\n' +
          '\n' +
          'The value at position 1 is not a member of ‘String’.\n');
 
-  throws(function() { S.get(Number, 'x', null); },
+  throws(function() { S.get(S.K(true), 'x', null); },
          TypeError,
          'Type-class constraint violation\n' +
          '\n' +
-         'get :: Accessible a => TypeRep -> String -> a -> Maybe b\n' +
-         '       ^^^^^^^^^^^^                         ^\n' +
-         '                                            1\n' +
+         'get :: Accessible a => Function -> String -> a -> Maybe c\n' +
+         '       ^^^^^^^^^^^^                          ^\n' +
+         '                                             1\n' +
          '\n' +
          '1)  null :: Null\n' +
          '\n' +
          '‘get’ requires ‘a’ to satisfy the Accessible type-class constraint; the value at position 1 does not.\n');
 
-  eq(S.get(Number, 'x', {x: 0, y: 42}), S.Just(0));
-  eq(S.get(Number, 'y', {x: 0, y: 42}), S.Just(42));
-  eq(S.get(Number, 'z', {x: 0, y: 42}), S.Nothing);
-  eq(S.get(String, 'x', {x: 0, y: 42}), S.Nothing);
+  eq(S.get(S.K(true), 'x', {x: 0, y: 42}), S.Just(0));
+  eq(S.get(S.K(true), 'y', {x: 0, y: 42}), S.Just(42));
+  eq(S.get(S.K(true), 'z', {x: 0, y: 42}), S.Just(undefined));
```

Comment:
This behavior was discussed with @davidchambers on Gitter, and he noted that he would prefer this to return `Nothing` (by determining beforehand whether the object property even exists). This has several implications:

* The behavior will change from how it currently works. Currently; if we were to provide a `TypeRep Nil`, we would correctly get `Just(undefined)`. The only reason we don't come across this is because `TypeRep Nil` does not exist.
* We will have to make a decision on what constitutes existence of an object property, eg. do we walk the prototype chain? Currently we delegate all of the decision making to the type identifier.
* I'm concerned that behavior might be unexpected. If I provide a type identifier which includes `undefined` as one of its members eg. `union([Nil, String])` then it would behave the same as `String` for JSON parsed data.

What are your thoughts?